test-framework/common/src/main/java/io/quarkus/test/common/TestResourceManager.java

```diff
@@ -46,17 +64,41 @@ public void init() {
 
     public Map<String, String> start() {
         started = true;
-        Map<String, String> ret = new HashMap<>();
-        for (TestResourceEntry entry : testResourceEntries) {
-            try {
-                Map<String, String> start = entry.getTestResource().start();
-                if (start != null) {
-                    ret.putAll(start);
+        Map<String, String> ret = new ConcurrentHashMap<>();
+
+        ExecutorService executor = newExecutor();
```

Comment:
This needs to be shutdown in a finally block

Reply:
Well spotted! Thanks :)
I'll fix it along with the other issue you mentioned after you respond